inc/integrations/class-integrations-manager.php

```diff
@@ -27,19 +30,82 @@ class Integrations_Manager {
 	public static function instance() {
 		if ( ! isset( static::$instance ) ) {
 			static::$instance = new static();
-			static::$instance->setup();
 		}
 		return static::$instance;
 	}
 
 	/**
-	 * Class constructor.
+	 * Instantiate each integration's class instance and run through its setup procedures.
 	 */
-	public function setup() {
+	public function setup_integrations() {
 		// Register admin page.
 		add_action( 'admin_menu', [ $this, 'register_admin' ] );
 		// Register settings fields for integrations.
-		add_action( 'admin_init', [ $this, 'register_settings_fields' ], 10, 2 );
+		add_action( 'admin_init', [ $this, 'register_settings_fields' ] );
+
+		// Archiveless.
```

Comment:
This totally works for now. Could be cool to have an array of integration classes that are supported out of the box and then have one place that loops through all enabled integrations and loads them, something like this (pseudocode):

```php
$integrations = [
	'archiveless' => __NAMESPACE__ . '\\Archiveless',
	'google_analytics' => __NAMESPACE__ . '\\ Google_Analytics',
	//etc.
];

// Allow new integrations to be added via a filter.
$integrations = apply_filters( 'wp_irving_active_integrations', $integrations );

foreach ( $integrations as $type => $class ) {
	$integration = new $class();
	$integration->setup();
}
```

That would DRY out this method a bit and allow for projects to add custom integrations without Irving Core providing them. We could even write an interface for Integrations to help teams conform to the requirements of the manager.